fakestorage/bucket.go

```diff
@@ -70,6 +71,18 @@ func (s *Server) createBucketByPost(r *http.Request) jsonResponse {
 		return jsonResponse{errorMessage: err.Error(), status: http.StatusBadRequest}
 	}
 
+	_, err := s.backend.GetBucket(name)
```

Comment:
I was wondering if we should have the backend return an error or something like that (to avoid TOCTOU). This is an emulator though, I imagine TOCTOU isn't really an issue, and if it becomes an issue we can fix it x)